Userland/Libraries/LibWeb/HTML/HTMLScriptElement.cpp

```diff
@@ -435,12 +436,52 @@ void HTMLScriptElement::prepare_script()
         }
         // -> "importmap"
         else if (m_script_type == ScriptType::ImportMap) {
-            // FIXME: 1. If el's relevant global object's import maps allowed is false, then queue an element task on the DOM manipulation task source given el to fire an event named error at el, and return.
-
-            // FIXME: 2. Set el's relevant global object's import maps allowed to false.
-
-            // FIXME: 3. Let result be the result of creating an import map parse result given source text and base URL.
+            // 1. If el's relevant global object's import maps allowed is false, then queue an element task on the DOM manipulation task source given el to fire an event named error at el, and return.
+            if (!document().window().import_maps_allowed()) {
+                queue_an_element_task(HTML::Task::Source::DOMManipulation, [this] {
+                    dispatch_event(DOM::Event::create(realm(), HTML::EventNames::error));
+                });
+                return;
+            }
+            // 2. Set el's relevant global object's import maps allowed to false.
+            document().window().set_import_maps_allowed(false);
+            // 3. Let result be the result of creating an import map parse result given source text and base URL.
+
+            // https://html.spec.whatwg.org/multipage/webappapis.html#parse-an-import-map-string
```

Comment:
Please put the "parse an import map string" code in a separate function/method, so it's easier to follow.